spring-kafka/src/main/java/org/springframework/kafka/retrytopic/EndpointCustomizerFactory.java

```diff
@@ -63,41 +64,58 @@ public EndpointCustomizerFactory(DestinationTopic.Properties destinationProperti
 		this.retryTopicNamesProviderFactory = retryTopicNamesProviderFactory;
 	}
 
-	public final EndpointCustomizer createEndpointCustomizer() {
-		return addSuffixesAndMethod(this.destinationProperties, this.beanMethod.resolveBean(this.beanFactory),
-				this.beanMethod.getMethod());
+	public final EndpointCustomizer<MethodKafkaListenerEndpoint<?, ?>> createEndpointCustomizer() {
+		return addSuffixesAndMethod(this.destinationProperties);
 	}
 
-	protected EndpointCustomizer addSuffixesAndMethod(DestinationTopic.Properties properties, Object bean, Method method) {
+	protected EndpointCustomizer<MethodKafkaListenerEndpoint<?, ?>> addSuffixesAndMethod(
```

Comment:
Public API modified. It may theoretically break any external implementations (although very unlikely that some might be implementing this, we need to be mindful of that possibility. Why don't we deprecate the original and add the modification separately?